.storybook/preview.tsx

```diff
@@ -0,0 +1,154 @@
+import {
+  Global,
+  ThemeProvider,
+  themes,
+  createReset,
+  convert,
+  styled,
+  useTheme,
+} from "@storybook/theming";
+import type { Preview } from "@storybook/react";
+import { initialize, mswDecorator } from "msw-storybook-addon";
+import React from "react";
+
+// Initialize MSW
+initialize({
+  onUnhandledRequest(req) {
+    if (req.url.origin !== document.location.origin) {
+      console.error(
+        `[MSW] %s %s %s (UNHANDLED)`,
+        new Date().toTimeString().slice(0, 8),
+        req.method.toUpperCase(),
+        req.url.href
+      );
+    }
+  },
+});
+
+const Panels = styled.div<{ orientation: "right" | "bottom" }>(
+  ({ orientation }) => ({
+    flexDirection: orientation === "right" ? "row" : "column",
+  }),
+  {
+    width: "100vw",
+    height: "100vh",
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+    gap: 20,
+  }
+);
+
+const Panel = styled.div<{ orientation: "right" | "bottom" }>(
+  ({ orientation }) => ({
+    width: orientation === "right" ? 420 : 900,
+    height: orientation === "right" ? 640 : 300,
+    overflow: "auto",
+  }),
+  ({ theme }) => ({
+    position: "relative",
+    outline: `1px solid ${theme.color.border}`,
+    background: theme.background.content,
+    color: theme.color.defaultText,
+    fontSize: theme.typography.size.s2 - 1,
+  })
+);
+
+const ThemedSetRoot = () => {
+  const theme = useTheme();
+  React.useEffect(() => {
+    document.body.style.background = theme.background.content;
+    document.body.style.color = theme.color.defaultText;
+    document.body.style.fontSize = `${theme.typography.size.s2 - 1}px`;
+  });
+  return null;
+};
+
+export const decorators = [
+  // Provide the MSW addon decorator globally
+  mswDecorator,
+
+  // Render two panels side-by-side or stacked, depending on selected orientation
+  // Note this assumes any play functions are equipped to deal with multiple canvases
+  (StoryFn, { globals, parameters }) => {
```

Comment:
I've configured both the viewports addon and the theme switcher with custom decorator (the panels thing), just because the viewports addon would otherwise use the default mobile/desktop viewports, which doesn't seem very useful.

I didn't find an easy way to disable the viewports addon without disabling addon-essentials, so I kept it. We should really be using the theme switcher instead though.

Chromatic doesn't use the viewports configured for addon-viewports, it will simply snapshot the entire page with the side-by-side ("right") panel setup, since that's the default.